src/server/routes/api/document/reach.js

```diff
@@ -226,21 +236,107 @@ module.exports = {
               description: getSentenceText( frame.sentence )
             };
 
-            if( argsAreEnts ){
-              intn.entries = argFrames.map( getElFromFrame ).map( getEntryFromEl );
-            } else if( isBinaryCollapsible ){
```

Comment:
``getEntityFrame()`` was being called only if ``isBinaryCollapsible`` is true. I made a change to call it anyways since extra effort caused by it is only a cheap lookup check and we can avoid some code duplication by removing ``isBinaryCollapsible`` and ``argsAreEnts`` checks.

I can keep making ``isBinaryCollapsible`` and ``argsAreEnts`` checks if this is not okay. However, even in that case using these checks only to get something like ``entityFrames`` and making the remaining operations on ``entityFrames`` would make sense to avoid code duplication.

Reply:
@jvwong Do you remember why we restricted reach interactions to only controller-controlled.? I suspect it's to filter out interaction types we don't support, like complex formation.  If that's the case, we may need to keep isBinaryCollapsible in order to filter out unsupported types.

Reply:
@maxkfranz this code segment is inside ``if( argsAreEnts || isBinaryCollapsible )``. I think for the existing version of the code ``else if( isBinaryCollapsible )`` is functioning as if ``else``.

Reply:
If I recall correctly, it's important to keep it as we don't have a spec of the reach output.  We only support cases that we've manually analysed to make sure we convert accurately (even if we don't convert everything).

Reply:
Basically, `argsAreEnts || isBinaryCollapsible` means "can the reach output be converted to a binary interaction" -- either directly or indirectly.